okcupyd/filter.py

```diff
@@ -19,25 +18,26 @@ class Filters(object):
     def __init__(self):
         self.builders = []
         self.keys = set()
-        self._key_to_type = {}
-        self._key_to_values = {}
         self._key_to_string = {}
+        self._key_to_values = {}
+        self._key_to_type = {}
 
     @util.cached_property
     def filter_meta(filters_instance):
         class FilterMeta(util.decorate_all(staticmethod)):
 
-            acceptable_values = None
-            keys = ()
-            types = None
-            descriptions = None
             output_key = None
+            keys = () # must be a container
```

Comment:
That will work for class definitions and instance attributes, but not functions. I think that it can handle classes that are defined as closures so it would work in this case. See user.py for examples.